### PR TITLE
Build release native components on release build.

### LINKED ISSUE #8787: OuterLoop: Release builds test with previous build native binaries, causing upgrade failures
```
MESSAGE:
System.EntryPointNotFoundException : Unable to find an entry point named 'CryptoNative_EcKeyCreateByOid' in DLL 'System.Security.Cryptography.Native'.
+++++++++++++++++++
STACK TRACE:
at Interop.Crypto.EcKeyCreateByOid(String oid) at System.Security.Cryptography.ECDsaOpenSsl.GenerateKeyLazy() at System.Lazy`1.CreateValue() at System.Lazy`1.LazyInitValue() at System.Security.Cryptography.ECDsaOpenSsl.DuplicateKeyHandle() at System.Security.Cryptography.OpenSsl.Tests.RsaOpenSslTests.VerifyDuplicateKey_ECDsaHandle()
```

To see full test log [here](http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/master/job/outerloop_osx_release/47/#showFailuresLink)

### PR DESCRIPTION
**Pull request by Priya91**
cc @bartonjs

fixes #8787

### CODE REVIEW

**Comment by Priya91** [comment]
@dotnet-bot test outerloop osx release

File: netci.groovy [hunk 1]
```diff
@@ -331,10 +331,10 @@ def outerloopLinuxOSes = ['Ubuntu16.04', 'CentOS7.1', 'OpenSUSE13.2', 'RHEL7.2',
...
                     }
                     else if (os == 'OSX') {
-                        shell("HOME=\$WORKSPACE/tempHome ./build.sh /p:ConfigurationGroup=${configurationGroup} /p:Outerloop=true /p:TestWithLocalLibraries=true /p:WithoutCategories=IgnoreForCI")
+                        shell("HOME=\$WORKSPACE/tempHome ./build.sh ${configurationGroup.toLowerCase()} /p:ConfigurationGroup=${configurationGroup} /p:Outerloop=true /p:TestWithLocalLibraries=true /p:WithoutCategories=IgnoreForCI")
                     }
                     else {
-                        shell("sudo HOME=\$WORKSPACE/tempHome ./build.sh /p:ConfigurationGroup=${configurationGroup} /p:Outerloop=true /p:TestWithLocalLibraries=true /p:WithoutCategories=IgnoreForCI")
+                        shell("sudo HOME=\$WORKSPACE/tempHome ./build.sh ${configurationGroup.toLowerCase()} /p:ConfigurationGroup=${configurationGroup} /p:Outerloop=true /p:TestWithLocalLibraries=true /p:WithoutCategories=IgnoreForCI")
```
Comment (bartonjs) [inline netci.groovy]: Does it need both the toLowerCase and /p versions? I'd assume that the bareword sets the /p value.

**Comment by bartonjs** [comment]
cc: @ellismg

**Comment by ellismg** [comment]
So the only issue here is that if you make a change between the shim and the managed library together, I don't think we'll pick up the shim changes even for the test projects that logically depend on the shim.

File: netci.groovy (hunk 1 above)
Comment (Priya91) [inline netci.groovy]: Yeah it does, in build.sh, but this doesn't break anything..

**Comment by Priya91** [comment]
@ellismg I think that feedback was meant for this [PR](https://github.com/dotnet/corefx/pull/8893)

File: netci.groovy (hunk 1 above)
Comment (bartonjs) [inline netci.groovy]: I'm mainly concerned that someone in the future would see both, and think "oh, I bet I can remove one of these"... and things'll still look like they work.  If we remove the removable one now, there's no duplication.

If you don't think that's really a risk, then I'll accept that assessment.

**Comment by bartonjs** [comment]
The change looks reasonable; but I don't know enough to say if it's gone far enough.

LGTM on the basis that you know more about this than I do :smile:.

**Comment by ellismg** [comment]
@Priya91 Sorry for commenting on the wrong PR.  This all sounds good to me.

**Comment by Priya91** [comment]
Thanks for the review ..

**Pull request merged**